src/ObjectUtils/PdbFileDia.cpp

```diff
@@ -84,47 +87,149 @@ ErrorMessageOr<void> PdbFileDia::LoadDataForPDB() {
-    } else {
-      ORBIT_ERROR("Unable to retrieve parameter types of function %s. Error: %s",
-                  symbol_info.demangled_name(), parameter_list_or_error.error().message());
-    }
-    SysFreeString(function_name);
-
-    DWORD relative_virtual_address = 0;
-    if (dia_symbol->get_relativeVirtualAddress(&relative_virtual_address) != S_OK) continue;
-    symbol_info.set_address(relative_virtual_address + object_file_info_.load_bias);
-
-    ULONGLONG length = 0;
-    if (dia_symbol->get_length(&length) != S_OK) continue;
-    symbol_info.set_size(length);
-
-    *module_symbols.add_symbol_infos() = std::move(symbol_info);
+    std::invoke(std::forward<Consumer>(consumer), dia_symbol);
   }
 
+  return outcome::success();
+}
+}  // namespace
+
+ErrorMessageOr<orbit_grpc_protos::ModuleSymbols> PdbFileDia::LoadDebugSymbols() {
+  ModuleSymbols module_symbols;
+  absl::flat_hash_set<uint64_t> addresses_from_module_info_stream;
+
+  // Find the function symbols in the module info stream. For now, we ignore "blocks" and
+  // "thunks". "Thunks" (which are 5 byte long jumps from incremental linking) don't even
+  // have a name, and while "blocks" (nested scopes inside functions) may have names
+  // according to the documentation, we have never observed that in real PDB files.
+  OUTCOME_TRY(ForEachSymbolWithSymTag(
+      SymTagFunction, dia_global_scope_symbol_.p, file_path_.string(),
+      [&module_symbols, &addresses_from_module_info_stream, this](IDiaSymbol* dia_symbol) {
+        SymbolInfo symbol_info;
+
+        BSTR function_name = {};
+        if (dia_symbol->get_name(&function_name) != S_OK) return;
+        symbol_info.set_demangled_name(orbit_base::ToStdString(function_name));
+        ErrorMessageOr<std::string> parameter_list_or_error =
+            PdbDiaParameterListAsString(dia_symbol);
+        if (parameter_list_or_error.has_value()) {
+          symbol_info.set_demangled_name(symbol_info.demangled_name() +
+                                         parameter_list_or_error.value());
+        } else {
+          ORBIT_ERROR("Unable to retrieve parameter types of function %s. Error: %s",
+                      symbol_info.demangled_name(), parameter_list_or_error.error().message());
+        }
+        SysFreeString(function_name);
+
+        DWORD relative_virtual_address = 0;
+        if (dia_symbol->get_relativeVirtualAddress(&relative_virtual_address) != S_OK) return;
+        symbol_info.set_address(relative_virtual_address + object_file_info_.load_bias);
+
+        ULONGLONG length = 0;
+        if (dia_symbol->get_length(&length) != S_OK) return;
+        symbol_info.set_size(length);
+
+        addresses_from_module_info_stream.insert(symbol_info.address());
+        *module_symbols.add_symbol_infos() = std::move(symbol_info);
+      }));
+
+  // Check the public symbol stream for additional function symbols. Many public
+  // symbols are already defined in the module info stream, so we will skip those
```

Comment:
Do we know if symbols with the same address actually are equal between the two streams? (For our purposes: same size, same name.)

Reply:
Yes and no. At least I looked into this:

1. The size is equal, as far as I have seen.
2. The name is in a mangled form, and is not equal.
    a) In most of the cases, it is effectively the "same" name
    b) There are some cases where (already) in the module symbols, the same symbol has multiple names. As far as I have seen it, for those cases we only have one public symbol.